Sources/News.php

```diff
@@ -1760,101 +1895,884 @@ function getXmlProfile($xml_format)
+					),
+					array(
+						'tag' => 'modified_by',
+						'attributes' => !empty($row['modified_name']) ? array('label' => $txt['modified_by']) : null,
+						'content' => !empty($row['modified_name']) ? $row['modified_name'] : null,
+						'cdata' => true,
+					),
+					array(
+						'tag' => 'modified_reason',
+						'attributes' => !empty($row['modified_reason']) ? array('label' => $txt['reason_for_edit']) : null,
+						'content' => !empty($row['modified_reason']) ? $row['modified_reason'] : null,
+						'cdata' => true,
+					),
+					array(
+						'tag' => 'likes',
+						'attributes' => array('label' => $txt['likes']),
+						'content' => $row['likes'],
+					),
+					array(
+						'tag' => 'approval-status',
+						'attributes' => $show_all ? array('label' => $txt['approval_status']) : null,
+						'content' => $show_all ? $row['approved'] : null,
+					),
+					array(
+						'tag' => 'attachments',
+						'attributes' => array('label' => $txt['attachments']),
+						'content' => $attachments,
+					),
+				),
+			);
+		}
+	}
+	$smcFunc['db_free_result']($request);
+
+	return $data;
+}
+
+/**
+ * Get a user's personal messages.
+ * Only the user can do this, and no one else -- not even the admin!
+ *
+ * @param string $xml_format The XML format. Can be 'atom', 'rdf', 'rss', 'rss2' or 'smf'
+ * @param bool $ascending If true, get the oldest PMs first. Default false.
+ * @return array An array of arrays containing data for the feed. Each array has keys corresponding to the appropriate tags for the specified format.
+ */
+function getXmlPMs($xml_format, $ascending = false)
+{
+	global $scripturl, $modSettings, $board, $txt, $context, $user_info;
+	global $query_this_board, $smcFunc, $sourcedir, $cachedir;
+
+	// Personal messages are supposed to be private
+	if (empty($context['xmlnews_uid']) || ($context['xmlnews_uid'] != $user_info['id']))
+		return array();
+
+	$select_id_members_to = $smcFunc['db_title'] === POSTGRE_TITLE ? "string_agg(pmr.id_member::text, ',')" : 'GROUP_CONCAT(pmr.id_member)';
+
+	$select_to_names = $smcFunc['db_title'] === POSTGRE_TITLE ? "string_agg(COALESCE(mem.real_name, mem.member_name), ',')" : 'GROUP_CONCAT(COALESCE(mem.real_name, mem.member_name))';
+
+	$request = $smcFunc['db_query']('', '
+		SELECT pm.id_pm, pm.msgtime, pm.subject, pm.body, pm.id_member_from, pm.from_name, ' . $select_id_members_to . ' AS id_members_to, ' . $select_to_names . ' AS to_names
+		FROM {db_prefix}personal_messages AS pm
+			INNER JOIN {db_prefix}pm_recipients AS pmr ON (pm.id_pm = pmr.id_pm)
+			INNER JOIN {db_prefix}members AS mem ON (mem.id_member = pmr.id_member)
+		WHERE (pm.id_member_from = {int:uid} OR pmr.id_member = {int:uid})
+			AND pm.id_pm > {int:start_after}
+		GROUP BY pm.id_pm
+		ORDER BY pm.id_pm {raw:ascdesc}
+		LIMIT {int:limit} OFFSET {int:offset}',
+		array(
+			'limit' => $context['xmlnews_limit'],
+			'offset' => !empty($context['pms_start']) ? 0 : $context['xmlnews_offset'],
```

Comment:
I believe the 4 references to 'pms_start' in this chunk of code are intended to be 'personal_messages_start'.  Making this change addresses the loop issue.

Reply:
Yes, indeed. Good catch, my friend!